File: template/src/main/java/ru/surfstudio/standard/ui/base/error/ErrorHandlerModule.java
package ru.surfstudio.standard.ui.base.error;

import dagger.Module;
import dagger.Provides;
import ru.surfstudio.android.core.app.dagger.scope.PerScreen;
import ru.surfstudio.android.core.ui.base.error.ErrorHandler;

@Module
public class ErrorHandlerModule {

    @Provides
    @PerScreen
    ErrorHandler provideNetworkErrorHandler(StandardErrorHandler standardErrorHandler){
        return standardErrorHandler;
    }
}

File: template/src/main/java/ru/surfstudio/standard/ui/base/error/ErrorHandlerModule.kt
package ru.surfstudio.standard.ui.base.error

import dagger.Module
import dagger.Provides
import ru.surfstudio.android.core.app.dagger.scope.PerScreen
import ru.surfstudio.android.core.ui.base.error.ErrorHandler

@Module
class ErrorHandlerModule {

    @Provides
    @PerScreen
    internal fun provideNetworkErrorHandler(standardErrorHandler: StandardErrorHandler): ErrorHandler {
        return standardErrorHandler
    }
}
